google-map.html

```diff
@@ -371,7 +371,7 @@
       if (this.map) {
         return; // already initialized
       }
-      if (!(window.google && window.google.maps)) {
+      if (this.$.api.libraryLoaded !== true) {
```

Comment:
I like libraryLoaded, it guarantees google.maps.Map by API contract. 

By testing for google.maps.Map you are exposing yourself to future bug when Maps API decides to load everything async, and some other part of the API you need has not loaded yet. (like markers).